lib/ssl/src/ssl.erl

```diff
@@ -1765,7 +1770,7 @@ Options specific to the server side, or with different semantics for the client
         common_option_cert() |
         {alpn_preferred_protocols,  AppProtocols::[binary()]}|
         {sni_hosts, SNIHosts::[{inet:hostname(), [server_option() | common_option()]}]} |
-        {sni_fun, SNIFun:: fun((string()) -> [])} |
+        {sni_fun, SNIFun:: fun((string()) -> [server_option() | common_option()] | undefined)} |
```

Comment:
`'unrecognized'` missing,  undefined should be quoted

Reply:
fixed